examples/Titanic/titanic_handler.py

```diff
@@ -0,0 +1,160 @@
+        self.predicted_idx = None
+        self.out_probs = None
+        self.delta = None
+        self.input_file_path = None
+
+    def initialize(self, ctx):
+        """In this initialize function, the Titanic trained model is loaded and
+        the Integrated Gradients Algorithm for Captum Explanations
+        is initialized here.
+
+        Args:
+            ctx (context): It is a JSON Object containing information
+            pertaining to the model artifacts parameters.
+        """
+        self.manifest = ctx.manifest
+        properties = ctx.system_properties
+        model_dir = properties.get("model_dir")
+        print("Model dir is {}".format(model_dir))
+        serialized_file = self.manifest["model"]["serializedFile"]
+        model_pt_path = os.path.join(model_dir, serialized_file)
+        self.device = torch.device(
+            "cuda:" + str(properties.get("gpu_id")) if torch.cuda.is_available() else "cpu"
+        )
+
+        self.model = TitanicSimpleNNModel()
+        self.model.load_state_dict(torch.load(model_pt_path))
+        self.model.to(self.device)
+        self.model.eval()
+
+        logger.info("Titanic model from path %s loaded successfully", model_dir)
+
+        # Read the mapping file, index to object name
+        mapping_file_path = os.path.join(model_dir, "index_to_name.json")
+        if os.path.isfile(mapping_file_path):
+            print("Mapping file present")
+            with open(mapping_file_path) as f:
+                self.mapping = json.load(f)
+        else:
+            print("Mapping file missing")
+            logger.warning("Missing the index_to_name.json file.")
+
+        # ------------------------------- Captum initialization ----------------------------#
+        self.ig = IntegratedGradients(self.model)
+        self.initialized = True
+
+    def preprocess(self, data):
+        """Basic text preprocessing, based on the user's chocie of application mode.
+
+        Args:
+            data (csv): The Input data in the form of csv is passed on to the preprocess
+            function.
+
+        Returns:
+            list : The preprocess function returns a list of Tensor and feature names
+        """
+        body = data[0]["body"]
+        self.input_file_path = body["input_file_path"]
+        data = pd.read_csv(self.input_file_path)
+        self.feature_names = list(data.columns)
+        data = data.to_numpy()
+        data = torch.from_numpy(data).type(torch.FloatTensor)
+        return data, self.feature_names
+
+    def inference(self, data):
+        """Predict the class (survived or not survived) of the received input dataframe using the
+        serialized model.
+
+        Args:
+            input_batch (list): List of Text Tensors from the pre-process function is passed here
+
+        Returns:
+            list : It returns a list of the predicted value for the input text
+        """
+        self.out_probs = self.model(data[0]).detach().numpy()
```

Comment:
@Arvind-Ideas2IT setting device to GPU, this requires the data to be moved to right device. 
In case to prevent from converting it to cpu for np.argmax,  "self.out_probs.argmax(1).item()", could be used as well.

Reply:
@HamidShojanazeri  Thanks for the suggestion. I will add this